core/math/aabb.cpp

```diff
@@ -135,27 +134,30 @@ bool AABB::intersects_ray(const Vector3 &p_from, const Vector3 &p_dir, Vector3 *
 				return false;
 			}
 		} else { // ray not parallel to planes in this direction
-			c1[i] = (position[i] - p_from[i]) / p_dir[i];
-			c2[i] = (end[i] - p_from[i]) / p_dir[i];
+			real_t t1 = (position[i] - p_from[i]) / p_dir[i];
+			real_t t2 = (end[i] - p_from[i]) / p_dir[i];
 
-			if (c1[i] > c2[i]) {
-				SWAP(c1, c2);
+			if (t1 > t2) {
+				SWAP(t1, t2);
 			}
-			if (c1[i] > near) {
-				near = c1[i];
+
+			if (t1 > tmin) {
+				tmin = t1;
 				axis = i;
 			}
-			if (c2[i] < far) {
-				far = c2[i];
+
+			if (t2 < tmax) {
+				tmax = t2;
 			}
-			if ((near > far) || (far < 0)) {
+
+			if (tmin > tmax) {
```

Comment:
Again here, if you can clarify why removing the `|| (far < 0)` is needed that would be useful for reviewing.

Reply:
see above